module-system/module-implementations/sov-ibc/src/context.rs

```diff
@@ -221,105 +370,216 @@ where
+            .packet_ack_map
+            .delete(ack_path, *self.working_set.borrow_mut());
+        Ok(())
     }
 
     fn store_channel(
         &mut self,
         channel_end_path: &ChannelEndPath,
         channel_end: ChannelEnd,
     ) -> Result<(), ContextError> {
-        todo!()
+        self.ibc.channel_end_map.set(
+            channel_end_path,
+            &channel_end,
+            *self.working_set.borrow_mut(),
+        );
+        Ok(())
     }
 
     fn store_next_sequence_send(
         &mut self,
         seq_send_path: &SeqSendPath,
         seq: Sequence,
     ) -> Result<(), ContextError> {
-        todo!()
+        self.ibc
+            .send_sequence_map
+            .set(seq_send_path, &seq, *self.working_set.borrow_mut());
+        Ok(())
     }
 
     fn store_next_sequence_recv(
         &mut self,
         seq_recv_path: &SeqRecvPath,
         seq: Sequence,
     ) -> Result<(), ContextError> {
-        todo!()
+        self.ibc
+            .recv_sequence_map
+            .set(seq_recv_path, &seq, *self.working_set.borrow_mut());
+        Ok(())
     }
 
     fn store_next_sequence_ack(
         &mut self,
         seq_ack_path: &SeqAckPath,
         seq: Sequence,
     ) -> Result<(), ContextError> {
-        todo!()
+        self.ibc
+            .ack_sequence_map
+            .set(seq_ack_path, &seq, *self.working_set.borrow_mut());
+        Ok(())
     }
 
     fn increase_channel_counter(&mut self) -> Result<(), ContextError> {
-        todo!()
+        let next_channel_counter = self
+            .ibc
+            .channel_counter
+            .get(*self.working_set.borrow_mut())
+            .ok_or(ChannelError::Other {
+                description: "Channel counter not found".to_string(),
+            })?
+            .checked_add(1)
+            .ok_or(ChannelError::CounterOverflow)?;
+
+        self.ibc
+            .channel_counter
+            .set(&next_channel_counter, *self.working_set.borrow_mut());
+
+        Ok(())
     }
 
     fn emit_ibc_event(&mut self, event: IbcEvent) -> Result<(), ContextError> {
-        todo!()
+        // Note: as an interim solution, we transform IBC events into Tendermint
+        // events to simplify the conversion process, avoiding the need for
+        // converting individual IBC event types into a key-value pair of `&str`
+        let tm_event =
+            tendermint::abci::Event::try_from(event).map_err(|_| ClientError::Other {
+                description: "Failed to convert IBC event to Tendermint event".to_string(),
+            })?;
+
+        let event_attribute: Vec<String> = tm_event
+            .attributes
+            .into_iter()
+            .map(|attr| format!("{attr:?}"))
+            .collect();
+
+        self.working_set
+            .borrow_mut()
+            .add_event(tm_event.kind.as_str(), event_attribute.join(",").as_str());
+
+        Ok(())
     }
 
     fn log_message(&mut self, message: String) -> Result<(), ContextError> {
```

Comment:
On a second look, I'm not sure this is the right implementation, as it stores log messages forever in the application. We should look into how "logs" are supposed to work in the Sovereign SDK

Reply:
I think it's better to leave it as a `todo!()` until we know for sure how to do it. It's probably even better to just return `Ok(())` then this, as this leads to unbounded state growth

Reply:
Good point. I was looking into that.. though couldn’t find any interface accessible to modules either be able to cache logs or include them to the transaction receipts. 
Agreed. Okay

Reply:
[25b64e7](https://github.com/informalsystems/sovereign-sdk/pull/1/commits/25b64e76bd26a38c5800cdda7a09ac44747c3f46)